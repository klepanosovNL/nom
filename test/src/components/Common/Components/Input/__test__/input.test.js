import React from 'react';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Input } from '../Input';

afterEach(cleanup);

describe('Input"s tests', () => {
	it('Sent only type', () => {
		const tree = renderer.create(<Input type='text'></Input>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Sent nothing', () => {
		const tree = renderer.create(<Input></Input>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Sent type and additional props', () => {
		const tree = renderer
			.create(<Input type='text' required className='class-name'></Input>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
