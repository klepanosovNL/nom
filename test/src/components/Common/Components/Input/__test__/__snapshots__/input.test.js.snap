// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input"s tests Sent nothing 1`] = `<input />`;

exports[`Input"s tests Sent only type 1`] = `
<input
  type="text"
/>
`;

exports[`Input"s tests Sent type and additional props 1`] = `
<input
  className="class-name"
  required={true}
  type="text"
/>
`;
